Decompile cultists

rr- opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #1693.

- Venice masked goon
- Knife thrower
- Shotgun boss

rr- committed:
tr2/objects: split cultist module

rr- committed:
tr2: port Cultist1_Initialise

rr- committed:
tr2: port Cultist1_Control

rr- committed:
tr2: port Knife

rr- committed:
tr2: port Cultist2_Control

rr- committed:
tr2: port Cultist3_Initialise

rr- committed:
tr2: port Cultist3_Control

aredfan commented:
The knife thrower and the shotgun boss both LGTM. There's just one issue with the Venice masked goon.

- [x] If Lara gets close enough to the Venice masked goon, he'll start running around and bumping into her but without attacking. In OG (or dev snapshot), the goon bumps into Lara a few times too but then he'll start shooting at her again.
![20241219_221005_Bartolis_Hideout](https://github.com/user-attachments/assets/018381ee-d54b-465d-a0ca-5e6750456dd3)

lahm86 commented on src/tr2/game/objects/creatures/cultist_1.c:
```diff
@@ -0,0 +1,311 @@
+        return;
+    }
+
+    __int16 v9; // ax
```
This seems like a leftover from porting maybe?

rr- replied:
Thanks – fixed.

rr- committed:
tr2/objects: improve formatting

rr- commented:
> The knife thrower and the shotgun boss both LGTM. There's just one issue with the Venice masked goon.
> 
> * [ ]  If Lara gets close enough to the Venice masked goon, he'll start running around and bumping into her but without attacking. In OG (or dev snapshot), the goon bumps into Lara a few times too but then he'll start shooting at her again.
>   ![20241219_221005_Bartolis_Hideout](https://private-user-images.githubusercontent.com/94776985/397546470-018381ee-d54b-465d-a0ca-5e6750456dd3.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2OTEzNzMsIm5iZiI6MTczNDY5MTA3MywicGF0aCI6Ii85NDc3Njk4NS8zOTc1NDY0NzAtMDE4MzgxZWUtZDU0Yi00NjVkLWEwY2EtNWU2NzUwNDU2ZGQzLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDEwMzc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5ZjBhZjQ1YTM3Yjk2MTgwNzA4NTYzYTY3NjVkMGZlMDZlN2ZkOTU4YjYwOGI0YmRjMGE5MzQ2YjZmYWE5M2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mlhP0FlC0RFxkRj_82aArN23ycjh8US9LgRICwNSYOA)

Rebased on top of #2084 and fixed.
